src/tools/perf/lib/libperf_memory.c

```diff
@@ -41,6 +41,53 @@ static ucs_status_t ucp_perf_test_alloc_iov_mem(ucp_perf_datatype_t datatype,
     return UCS_OK;
 }
 
+static ucs_status_t ucp_perf_mem_alloc(const ucx_perf_context_t *perf,
+                                       size_t length,
+                                       ucs_memory_type_t mem_type,
+                                       void **address_p, ucp_mem_h *memh_p)
+{
+    ucp_mem_map_params_t params;
+    ucp_mem_attr_t attr;
+    ucs_status_t status;
+
+    params.field_mask  = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                         UCP_MEM_MAP_PARAM_FIELD_LENGTH |
+                         UCP_MEM_MAP_PARAM_FIELD_FLAGS |
+                         UCP_MEM_MAP_PARAM_FIELD_MEMORY_TYPE;
+    params.address     = NULL;
+    params.memory_type = mem_type;
+    params.length      = length;
+    params.flags       = UCP_MEM_MAP_ALLOCATE;
+    if (perf->params.flags & UCX_PERF_TEST_FLAG_MAP_NONBLOCK) {
+        params.flags |= UCP_MEM_MAP_NONBLOCK;
```

Comment:
Should we disallow MEM_MAP_NONBLOCKING with non-host mem_types or should we just let ucp_mem_map fail later in line 65?